### PR TITLE
Broken promises

### PR DESCRIPTION
**Pull request by dimensia**
Convert Tyranid to use newest MongoDB official node.js driver instead of promised-mongo.

### CODE REVIEW

**Comment by bsouthga** [comment]
such a depressing name for a branch

**Comment by yangchristian** [comment]
Saw some notes in the native API docs about deprecated methods. Not blocking for this PR, but I think it'd be good to add support for at least a few of the recommended replacements soon-ish so we can write up-to-date app code:

[findAndModify](http://mongodb.github.io/node-mongodb-native/2.1/api/Collection.html#findAndModify):

> use findOneAndUpdate, findOneAndReplace or findOneAndDelete instead

[insert](http://mongodb.github.io/node-mongodb-native/2.1/api/Collection.html#insert):

> Use insertOne, insertMany or bulkWrite

[update](http://mongodb.github.io/node-mongodb-native/2.1/api/Collection.html#update):

> use updateOne, updateMany or bulkWrite

[remove](http://mongodb.github.io/node-mongodb-native/2.1/api/Collection.html#remove):

> use deleteOne, deleteMany or bulkWrite

[save](http://mongodb.github.io/node-mongodb-native/2.1/api/Collection.html#save):

> use insertOne, insertMany, updateOne or updateMany

**Comment by bsouthga** [comment]
resolved a conflict with a small commit I made

File: src/core/collection.js [hunk 1]
```diff
@@ -464,57 +481,87 @@ export default class Collection {
...
+        }
+      });
+
+      return cursor;
+    }
+
+    if (auth) {
+      const secure = collection.secureQuery(args[0], 'view', auth);
+
+      if (secure.then) {
```
Comment (bsouthga) [inline src/core/collection.js:519]: smart, this seems like a good compromise without getting too deep in monkey patching

File: src/core/collection.js [hunk 2]
```diff
@@ -464,57 +481,87 @@ export default class Collection {
...
+  async findAll(...args) {
+    if (args.length === 1) {
+      const opts = args[0];
+
+      let v = opts.query;
+      if (v) {
+        const cursor = await this.find(v, opts.projection, opts);
+
+        if ( (v=opts.limit) ) {
+          cursor.limit(v);
```
Comment (bsouthga) [inline src/core/collection.js:544]: does this need to be `cursor = cursor.limit(v)`?

**Comment by bsouthga** [comment]
Looks good from my end! glad to be moving on from `promised-mongo`

**Pull request merged**